src/V4Router.sol

```diff
@@ -78,8 +79,9 @@ abstract contract V4Router is IV4Router, BaseActionsRouter, DeltaResolver {
             // Caching for gas savings
             uint256 pathLength = params.path.length;
             uint128 amountOut;
-            uint128 amountIn = params.amountIn;
             Currency currencyIn = params.currencyIn;
+            uint128 amountIn =
```

Comment:
lib could be used here too
so we dont have to repeat amount == OPEN_DELTA ? __ : ___